diff -Naur A/plat/imx/imx8qm/imx8qm_bl31_setup.c B/plat/imx/imx8qm/imx8qm_bl31_setup.c
--- A/plat/imx/imx8qm/imx8qm_bl31_setup.c	2020-11-10 10:43:14.186317279 +0530
+++ B/plat/imx/imx8qm/imx8qm_bl31_setup.c	2020-11-10 10:57:08.809150048 +0530
@@ -89,7 +89,8 @@
 	if (baudrate == 0)
 		panic();
 
-	sc_pm_get_clock_rate(ipc_handle, SC_R_UART_0, 2, &rate);
+	/* IWG27S: UART: LPUART4 Support for iWave platform */
+	sc_pm_get_clock_rate(ipc_handle, SC_R_UART_4, 2, &rate);
 
 	baud_diff = baudrate;
 	osr = 0;
@@ -424,14 +425,17 @@
 		panic();
 
 #if DEBUG_CONSOLE_A53
-	sc_pm_set_resource_power_mode(ipc_handle, SC_R_UART_0, SC_PM_PW_MODE_ON);
+	/* IWG27S: UART: LPUART4 Support for iWave platform */
+	sc_pm_set_resource_power_mode(ipc_handle, SC_R_UART_4, SC_PM_PW_MODE_ON);
 	sc_pm_clock_rate_t rate = 80000000;
-	sc_pm_set_clock_rate(ipc_handle, SC_R_UART_0, 2, &rate);
-	sc_pm_clock_enable(ipc_handle, SC_R_UART_0, 2, true, false);
+	/* IWG27S: UART: LPUART4 Support for iWave platform */
+	sc_pm_set_clock_rate(ipc_handle, SC_R_UART_4, 2, &rate);
+	sc_pm_clock_enable(ipc_handle, SC_R_UART_4, 2, true, false);
 
 	/* configure UART pads */
-	sc_pad_set(ipc_handle, SC_P_UART0_RX, UART_PAD_CTRL);
-	sc_pad_set(ipc_handle, SC_P_UART0_TX, UART_PAD_CTRL);
+	/* IWG27S: UART: LPUART4 Support for iWave platform */
+	sc_pad_set(ipc_handle, SC_P_M40_GPIO0_00, UART_PAD_CTRL);
+	sc_pad_set(ipc_handle, SC_P_M40_GPIO0_01, UART_PAD_CTRL);
 
 	lpuart32_serial_init(IMX_BOOT_UART_BASE);
 #endif
diff -Naur A/plat/imx/imx8qm/include/platform_def.h B/plat/imx/imx8qm/include/platform_def.h
--- A/plat/imx/imx8qm/include/platform_def.h	2020-11-10 10:43:14.186317279 +0530
+++ B/plat/imx/imx8qm/include/platform_def.h	2020-11-10 10:55:07.351862794 +0530
@@ -56,7 +56,8 @@
 #define PLAT_CCI_BASE			0x52090000
 #define CLUSTER0_CCI_SLVAE_IFACE	3
 #define CLUSTER1_CCI_SLVAE_IFACE	4
-#define IMX_BOOT_UART_BASE		0x5a060000
+/* IWG27S: UART: LPUART4 Support for iWave platform */
+#define IMX_BOOT_UART_BASE		0x5a0a0000
 #define IMX_BOOT_UART_BAUDRATE		115200
 #define IMX_BOOT_UART_CLK_IN_HZ		24000000
 #define PLAT_CRASH_UART_BASE		IMX_BOOT_UART_BASE
