diff -Naur A/iMX8M/soc.mak B/iMX8M/soc.mak
--- A/iMX8M/soc.mak	2022-07-04 18:07:07.839060709 +0530
+++ B/iMX8M/soc.mak	2022-07-04 18:09:14.530183584 +0530
@@ -161,7 +161,7 @@
 	./$(PAD_IMAGE) bl31.bin
 	./$(PAD_IMAGE) u-boot-nodtb.bin $(dtb)
 	DEK_BLOB_LOAD_ADDR=$(DEK_BLOB_LOAD_ADDR) TEE_LOAD_ADDR=$(TEE_LOAD_ADDR) ATF_LOAD_ADDR=$(ATF_LOAD_ADDR) ./mkimage_fit_atf.sh $(dtb) > u-boot-iwg.its
-	./mkimage_uboot -E -p 0x3000 -f u-boot-iwg.its u-boot-iwg.itb
+	mkimage -E -p 0x3000 -f u-boot-iwg.its u-boot-iwg.itb
 	@rm -f u-boot-iwg.its $(dtb)
 
 dtb_ddr3l = valddr3l.dtb
